app/Models/Service.php

```diff
@@ -0,0 +1,140 @@
+        $saturdayID = self::createWeekDay($businessHours[self::SATURDAY]);
+        $sundayID = self::createWeekDay($businessHours[self::SUNDAY]);
+
+        $sql = sprintf("INSERT INTO business_hours (%s, %s, %s, %s, %s, %s, %s)
+                        VALUES (:%s, :%s, :%s, :%s, :%s, :%s, :%s)",
+
+                        self::MONDAY, self::TUESDAY, self::WEDNESDAY, self::THURSDAY, self::FRIDAY, self::SATURDAY, self::SUNDAY,
+                        self::MONDAY, self::TUESDAY, self::WEDNESDAY, self::THURSDAY, self::FRIDAY, self::SATURDAY, self::SUNDAY);
+
+        return self::exeSQL($sql, [
+            self::MONDAY    => self::createBusinessHour($mondayID),
+            self::TUESDAY   => self::createBusinessHour($tuesdayID),
+            self::WEDNESDAY => self::createBusinessHour($wednesdayID),
+            self::THURSDAY  => self::createBusinessHour($thursdayID),
+            self::FRIDAY    => self::createBusinessHour($fridayID),
+            self::SATURDAY  => self::createBusinessHour($saturdayID),
+            self::SUNDAY    => self::createBusinessHour($sundayID)
+        ]);
+    }
+
+    private static function createBusinessHour(string | null $id): array
+    {
+        return [ $id, $id ? PDO::PARAM_INT : PDO::PARAM_NULL ];
+    }
+
+    private static function createWeekDay(array | null $period): int | null
+    {
+        if( $period === null ) return null;
+
+        $startTime = $period['starts'];
+        $endTime = $period['ends'];
+
+        self::checkTimeValidity($startTime, $endTime);
+
+        $sql = "INSERT INTO week_day (start_time, end_time)
+                VALUES (:start_time, :end_time)";
+
+		return self::exeSQL($sql, [
+            'start_time' => [ $startTime, PDO::PARAM_STR ],
+            'end_time'   => [ $endTime,   PDO::PARAM_STR ]
+        ]);
+    }
+
+    private static function checkTimeValidity(string $startTime, string $endTime): void
+    {
+        if($endTime < $startTime)
+            throw new Exception("End hours of work should be after start hours");
+    }
+
+    private static function checkDurationValidity(int $duration): void {
+        if ($duration < 60)
+            throw new Exception("Duration can't be shorter than a minute");
+    }
+
+    private static function createBuffer(int $beforeTime, int $afterTime): int
+    {
+        $sql = "INSERT INTO buffer (before_time, after_time)
+                VALUES (:before_time, :after_time)";
+
+        return self::exeSQL($sql, [
+            'before_time' => [ $beforeTime, PDO::PARAM_INT ],
+            'after_time'  => [ $afterTime,  PDO::PARAM_INT ]
+        ]);
+    }
+
+    private static function exeSQL(string $sql, array $valuesNTypes): int
+    {
+        $stmt = self::getConnection()->prepare($sql);
+
+        foreach ($valuesNTypes as $k => $v) {
+            $stmt->bindValue(":$k", $v[0], $v[1]);
+        }
+
+		$stmt->execute();
+
+		return self::getConnection()->lastInsertId();
+    }
```

Comment:
bu method da service classinin icinde olmali deyil normalda, hansisa traitin icinde ola biler, Model classi da hemin traiti use eder. Bele olsa avtomatik butun modeller istifade ede biler onu. Ve ya DB classinin methodu ola biler, DB::exeSQL() formatinda.

Reply:
DB class -na el vurmaq istemedim. PHP ise multiple inheritance desteklemir ve trait ne oldugunu indi oyrendim. PHP -ni on the fly oyrenirem evvel yazmamisam. Duzelderem